aws/resource_aws_lambda_permission.go

```diff
@@ -301,6 +301,13 @@ func resourceAwsLambdaPermissionDelete(d *schema.ResourceData, meta interface{})
 	log.Printf("[DEBUG] Removing Lambda permission: %s", input)
 	_, err := conn.RemovePermission(&input)
 	if err != nil {
+		// Missing whole policy or Lambda function (API error)
+		if awsErr, ok := err.(awserr.Error); ok {
+			if awsErr.Code() == "ResourceNotFoundException" {
```

Comment:
Nit: We prefer to use our helper function to determine these and use the SDK provided constants:

```go
if isAWSErr(err, lambda.ErrCodeResourceNotFoundException, "") {
```